src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -295,6 +295,34 @@ public IEnumerable<UnspentOutputReference> GetAllSpendableTransactions(CoinType
 
             return accounts.SelectMany(x => x.GetSpendableTransactions(currentChainHeight, this.Network.Consensus.CoinbaseMaturity, confirmations));
         }
+
+        /// <summary>
+        /// Finds the HD addresses for the public address.
+        /// </summary>
+        /// <remarks>
+        /// Returns an HDAddress.
+        /// </remarks>
+        /// <param name="coinType">Type of the coin to get the HD Address from.</param>
+        /// <param name="externalAddress">An address.</param>
+        /// <param name="accountFilter">An optional filter for filtering the accounts being returned.</param>
+        /// <returns>HD Address</returns>
+        public HdAddress FindHDAddressByExternalAddress(CoinType coinType, string externalAddress, Func<HdAccount, bool> accountFilter = null)
+        {
+            Guard.NotNull(externalAddress, nameof(externalAddress));
+
+            HdAccount account = this.GetAccountsByCoinType(coinType, accountFilter).FirstOrDefault(x => x.ExternalAddresses.Select(a => a.Address).Contains(externalAddress));
```

Comment:
Using `FirstOrDefault` means we expect that account can be `null` here, but do not check for this.

Reply:
Hi, I do check for a `null` on line 318

Reply:
That check is for the `HdAddress` being null, not for `HdAccount`.

Reply:
I just deleted my comment hoping you didn't get it yet.. haha Thanks.

Reply:
Alright, I found a much cleaner way to do this with built in methods.. Thanks for your suggestion... Will submit changes.